tests/test_edgeql_functions.py

```diff
@@ -8308,3 +8308,2149 @@ async def test_edgeql_functions_complex_types_04(self):
+                create required property a -> int64;
+            };
+            insert Bar{a := 1};
+            insert Bar{a := 2};
+            insert Bar{a := 3};
+            create function foo(x: int64) -> optional Bar {
+                set is_inlined := true;
+                using ((select Bar{a} filter .a = x limit 1));
+            };
+        ''')
+        await self.assert_query_result(
+            'select foo(<int64>{}).a',
+            [],
+        )
+        await self.assert_query_result(
+            'select foo(-1).a',
+            [],
+        )
+        await self.assert_query_result(
+            'select foo(1).a',
+            [1],
+        )
+        await self.assert_query_result(
+            'select foo({1, 2, 3}).a',
+            [1, 2, 3],
+            sort=True,
+        )
+
+    async def test_edgeql_functions_inline_object_02(self):
+        await self.con.execute('''
+            create type Bar {
+                create required property a -> int64;
+            };
+            insert Bar{a := 1};
+            insert Bar{a := 2};
+            insert Bar{a := 3};
+            create function foo(x: Bar) -> Bar {
+                set is_inlined := true;
+                using (x);
+            };
+        ''')
+        await self.assert_query_result(
+            'select foo(<Bar>{}).a',
+            [],
+        )
+        await self.assert_query_result(
+            'select foo((select Bar filter .a = 1)).a',
+            [1],
+        )
+        await self.assert_query_result(
+            'select foo((select Bar)).a',
+            [1, 2, 3],
+            sort=True,
+        )
+
+    async def test_edgeql_functions_inline_object_03(self):
+        await self.con.execute('''
+            create type Bar {
+                create required property a -> int64;
+            };
+            insert Bar{a := 1};
+            insert Bar{a := 2};
+            insert Bar{a := 3};
+            create function foo(x: optional Bar) -> optional Bar {
+                set is_inlined := true;
+                using (x ?? (select Bar filter .a = 1 limit 1));
+            };
+        ''')
+        await self.assert_query_result(
+            'select foo(<Bar>{}).a',
+            [1],
+        )
+        await self.assert_query_result(
+            'select foo((select Bar filter .a = 1)).a',
+            [1],
+        )
+        await self.assert_query_result(
+            'select foo((select Bar)).a',
+            [1, 2, 3],
+            sort=True,
+        )
+
+    async def test_edgeql_functions_inline_object_04(self):
+        await self.con.execute('''
+            create type Bar {
+                create required property a -> int64;
+            };
+            insert Bar{a := 1};
+            insert Bar{a := 2};
+            insert Bar{a := 3};
+            create function foo(x: Bar) -> int64 {
+                set is_inlined := true;
+                using (x.a);
+            };
+        ''')
```

Comment:
Could we do a test like
```
            create function foo() -> int64 {
                set is_inlined := true;
                using (count(Bar));
            };
```
and then query
```
select (Bar, foo())
```
and make sure that we get `3` back as the result from `foo()`.
I want to make sure that paths in the inlined functions don't end up getting captured by factoring with the enclosing query.